packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -248,11 +247,12 @@ abstract contract FundingRateApplier is FeePayer {
                 // Set proposal time to 0 since this proposal has now been resolved.
                 fundingRate.proposalTime = 0;
             } catch {
-                // Stop tracking if in dispute to allow other proposals to come in.
-                if (
-                    optimisticOracle.getRequest(address(this), identifier, proposalTime, ancillaryData).disputer !=
-                    address(0)
-                ) {
+                // Stop tracking and allow other proposals to come in if:
+                // - The requester address is empty, indicating that the Oracle does not know about
+                //   this funding rate request. This is possible if the Oracle is replaced while the price request
+                //   is still pending.
+                // - The request has been disputed.
+                if (request.disputer != address(0) || request.proposer == address(0)) {
```

Comment:
what are your thoughts on comparing this to `oo.getState` where `state != invalid or state != Settled` or something to that effect

Reply:
This is an interesting idea. I think comparing the state would be more readable, but would add more bytes because we'd still have to make the `getRequest` call to determine who the original proposer was (to potentially pay rewards to). WDYT?

Reply:
yea again easier to read but less optimized! This really should be directed at the OO interface to include current state on the request. I think leave it as is.